batch/batch/driver/instance_collection/pool.py

```diff
@@ -411,11 +411,21 @@ async def user_runnable_jobs(user, remaining):
                 (user,),
                 "user_runnable_jobs__select_running_batches",
             ):
+                # TODO Probably need to index jobs.update_id
                 async for record in self.db.select_and_fetchall(
                     '''
+SELECT job_id, spec, cores_mcpu
+FROM batch_updates
+STRAIGHT_JOIN jobs FORCE INDEX(jobs_batch_id_state_always_run_inst_coll_cancelled)
+ON batch_updates.batch_id = jobs.batch_id AND batch_updates.update_id = jobs.update_id
+WHERE batch_updates.batch_id = %s AND batch_updates.committed AND jobs.batch_id = %s AND state = 'Ready' AND always_run = 1 AND inst_coll = %s
+LIMIT %s;
+
 SELECT job_id, spec, cores_mcpu
 FROM jobs FORCE INDEX(jobs_batch_id_state_always_run_inst_coll_cancelled)
```

Comment:
The EXPLAIN for this query:

```
mysql> EXPLAIN SELECT job_id, spec, cores_mcpu
    ->  FROM jobs FORCE INDEX(jobs_batch_id_state_always_run_cancelled)
    ->  INNER JOIN batch_updates
    ->  ON jobs.batch_id = batch_updates.batch_id AND jobs.update_id = batch_updates.update_id
    ->  WHERE jobs.batch_id = 1000000 AND state = 'Ready' AND always_run = 0 AND inst_coll = "standard" AND cancelled = 0 AND batch_updates.committed
    ->  LIMIT 5;
+----+-------------+---------------+------------+--------+---------------------------------------------------------------------+------------------------------------------+---------+----------------------------+------+----------+-------------+
| id | select_type | table         | partitions | type   | possible_keys                                                       | key                                      | key_len | ref                        | rows | filtered | Extra       |
+----+-------------+---------------+------------+--------+---------------------------------------------------------------------+------------------------------------------+---------+----------------------------+------+----------+-------------+
|  1 | SIMPLE      | jobs          | NULL       | ref    | jobs_batch_id_state_always_run_cancelled                            | jobs_batch_id_state_always_run_cancelled | 132     | const,const,const,const    |    1 |    10.00 | Using where |
|  1 | SIMPLE      | batch_updates | NULL       | eq_ref | PRIMARY,batch_id,batch_updates_committed,batch_updates_start_job_id | PRIMARY                                  | 12      | const,batch.jobs.update_id |    1 |    90.00 | Using where |
+----+-------------+---------------+------------+--------+---------------------------------------------------------------------+------------------------------------------+---------+----------------------------+------+----------+-------------+
2 rows in set, 1 warning (0.03 sec)
```

Reply:
I think we should use this query instead of https://github.com/hail-is/hail/pull/12154/commits/0ab44dcf14ea7f542661b43acda67f020da9a638#r973232500 (the one above)